deploy/firesim

```diff
@@ -119,6 +119,16 @@ def buildafi(globalbuildconf):
 
     auto_create_bucket(globalbuildconf.s3_bucketname)
 
+    client = boto3.client('sns')
+
+    aws_resource_names_dict = aws_resource_names()
+    snsname = aws_resource_names_dict['snsname']
+
+    # this will either create the topic, if it doesn't exist, or just get the arn
+    client.create_topic(
+        Name=snsname
+    )
+
```

Comment:
1. You should capture this in a new function in https://github.com/sifive/firesim/blob/Topic_permission_check/deploy/awstools/awstools.py and refactor the other two uses to also call the new function:
   * https://github.com/sifive/firesim/blob/fafb94de6926350aebe7cd1f4bea63fc840312a3/deploy/awstools/awstools.py#L349-L358
   * https://github.com/sifive/firesim/blob/fafb94de6926350aebe7cd1f4bea63fc840312a3/deploy/awstools/awstools.py#L375-L385
   
2. You should put this in a try/except block with a helpful error message sent to `rootLogger.critical` before an empty `raise` to rethrow the exception.

Reply:
> refactor the other two uses to also call the new function
I wrapped the topic creation in a function.
> You should put this in a try/except block with a helpful error message
Created the same and to exit after the error.